### PR TITLE
add sign out convience api

### PR DESCRIPTION
**Pull request by FeiChen-okta**
#### Description:
Helper API that does sign out by revoking tokens, signing out of browser session and
remove persistent data.

#### Testing details:
- [x]  Verified basic functionality of change
- [x]  Added tests 

##### RESOLVES: 
[OKTA-239591](https://oktainc.atlassian.net/browse/OKTA-239591)

### CODE REVIEW

**Comment by IldarAbdullin-okta** [comment]
Readme update?

**Comment by FeiChen-okta** [comment]
> Readme update?

Will add to readme when releasing next version which is 1.0.4

File: library/src/main/java/com/okta/oidc/clients/web/WebAuthClient.java [hunk 1]
```diff
@@ -156,4 +157,49 @@ void registerCallback(ResultCallback<AuthorizationStatus, AuthorizationException
...
     void handleActivityResult(int requestCode, int resultCode, Intent data);
+
+    /**
+     * Convenience method to completely sign out of application.
+     * Performs the following operations in order:
+     * 1. Revokes the access_token. If this fails step 3 will not be attempted.
+     * 2. Revokes the refresh_token. If this fails step 3 will not be attempted.
+     * 3. Removes all persistence data. Only if steps 1 & 2 succeeds.
+     * 4. Clears browser session only if this is a {@link com.okta.oidc.clients.web.WebAuthClient}
+     * instance. Will always be attempted regardless of the other operations.
```
Comment (IldarAbdullin-okta) [inline library/src/main/java/com/okta/oidc/clients/web/WebAuthClient.java]: Is it the same as in js implementation? Or it is android specific implementation?

Comment (FeiChen-okta) [reply library/src/main/java/com/okta/oidc/clients/web/WebAuthClient.java]: It is the same as the js implementation. 

https://github.com/okta/oss-technical-designs/blob/master/technical_designs/rp-initiated-logout.md#mobile

**Pull request merged**